trio_websocket/_impl.py

```diff
@@ -808,6 +811,8 @@ def __str__(self):
 
         :param event:
         '''
+        self._for_testing_peer_closed_connection.set()
+        await trio.sleep(0)
```

Comment:
the checkpoint could be eliminated in the common case at the cost of more complexity

```python3
if self._for_testing_peer_closed_connection_enable:
    self._for_testing_peer_closed_connection.set()
    await trio.sleep(0)
```